sys/net/if_dl.h

```diff
@@ -64,7 +64,7 @@ struct sockaddr_dl {
 	u_char	sdl_nlen;	/* interface name length, no trailing 0 reqd. */
 	u_char	sdl_alen;	/* link level address length */
 	u_char	sdl_slen;	/* link layer selector length */
-	char	sdl_data[46];	/* minimum work area, can be larger;
+	char	sdl_data[46] __no_subobject_bounds;	/* minimum work area, can be larger;
```

Comment:
We probably want an annotation for "this is a custom flexible member thing with non-zero minimum length"... but also this is a highly dodgy API that should probably be sanitised upstream

Reply:
sockaddr in general is quite annoying for subobject bounds because they "inherit" from sockaddr which itself has a non-zero minimum length flexible member. Arguably though this annotation should be `__subobject_variable_length`.